python/ray/autoscaler/_private/gcp/config.py

```diff
@@ -30,6 +30,8 @@
 }
 
 # Those roles will be always added.
+# NOTE: `serviceAccountUser` allows the head node to create workers with
+# a serviceAccount. `roleViewer` allows the head node to run bootstrap_gcp.
```

Comment:
These changes look for me and I I verified that adding the role `roleViewer` works when running `bootsrap_gcp`.

The only nitpick is that these two commits could be squashed into a single one before merging this PR

Reply:
Yep! They will be squashed before merging :)